hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/FindSourceGreedy.java

```diff
@@ -49,7 +49,11 @@
       double currentUsageOfB = b.calculateUtilization(
           -sizeLeavingNode.get(b.getDatanodeDetails()));
       //in descending order
-      return Double.compare(currentUsageOfB, currentUsageOfA);
+      int ret = Double.compare(currentUsageOfB, currentUsageOfA);
+      if (ret != 0) {
+        return ret;
+      }
+      return a.hashCode() - b.hashCode();
```

Comment:
Can we do datanode details UUID comparison instead?

Reply:
sure, will do this.